src/vs/platform/terminal/common/terminal.ts

```diff
@@ -600,6 +601,7 @@ export interface ITerminalProcessOptions {
 	};
 	windowsEnableConpty: boolean;
 	environmentVariableCollections: ISerializableEnvironmentVariableCollections | undefined;
+	workspaceFolder: IWorkspaceFolder | undefined;
```

Comment:
```suggestion
	workspaceFolder?: IWorkspaceFolder;
```

then you can remove all of the `environmentVariableCollections: undefined`

Reply:
We should not miss to pass `workspaceFolder` when using `ITerminalProcessOptions`, so I don't think making it optional for sake of tests is the best approach.